File: solutions/typescript/leap/1/leap.ts
export function isLeap(year: number): boolean {
  // A year is a leap year if it's divisible by 4
  // BUT not if it's divisible by 100, unless it's also divisible by 400
  return (year % 4 === 0) && (year % 100 !== 0 || year % 400 === 0);
}

File: solutions/java/leap/1/src/main/java/Leap.java
 class Leap {

    boolean isLeapYear(int year) {
        // A leap year is divisible by 4,
        // unless it is also divisible by 100 (not a leap year),
        // unless it is also divisible by 400 (is a leap year).
        return (year % 4 == 0) && ((year % 100 != 0) || (year % 400 == 0));
    }

}